libraries/concurrent-merkle-tree/src/concurrent_merkle_tree.rs

```diff
@@ -46,6 +46,7 @@ fn check_leaf_index(leaf_index: u32, max_depth: usize) -> Result<(), ConcurrentM
 /// An additional key property of ConcurrentMerkleTree is support for [append](ConcurrentMerkleTree::append) operations,
 /// which do not require any proofs to be passed. This is accomplished by keeping track of the
 /// proof to the rightmost leaf in the tree (`rightmost_proof`).
+#[repr(C)]
```

Comment:
Btw, looks like you were caught by `unsafe` code!

Reply:
But I'm curious what the trigger was for completely different internal representation? Was it `rustc` change / upgrade?

Reply:
Yeah it could be, we'd have to look at the byte representation of old and new to check.